### DISCUSSION TITLE
Any way to avoid keeping buffers opened from `Go to definition` and similar ?

### DISCUSSION THREAD

**Discussion by Strackeror** [LazyVim/LazyVim#4874] [Q&A]
Heyo, transitioning from vscode to neovim, and i find myself missing something I found very useful in vscode which is the 'preview' editor, which reserves one and only one editor for peeking purposes until you actually edit a file.

I very often use go to definition to just peek at a file, then <C-o> to go back, but that very quickly ends up polluting the bufferline with a lot of files that are unlikely to stay relevant. I can delete the buffer immediately instead of jumping backward, but I'd rather have the consistency of jumping backward working in both cases if possible. Have I missed any obvious way to solve this problem ?

**Comment by zvrkSam** [comment]
Instead of buffers try opening in a split like so:
```lua
vim.keymap.set("n", "g|", "<cmd>vertical winc ]<CR>", { desc = "Go to vertical defenition", silent = true })
vim.keymap.set("n", "g-", "<cmd>horizontal winc ]<CR>", { desc = "Go to horizontal defenition", silent = true })
```
Also I suggest binding `:bdelete<CR>` to something simple like `<leader>d`
